From 98c45e683e640459e30d62d7d6d8cdf0066b8747 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Wed, 20 Aug 2008 01:30:59 +0200
Subject: [PATCH] Call InitErrorLog() right after BecomeDaemon()

The latter points stderr to /dev/null, so the sooner we open the log
after that, the better.
---
 dm.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/dm.c b/dm.c
index 52b49eb..015ce31 100644
--- a/dm.c
+++ b/dm.c
@@ -145,6 +145,8 @@ main (int argc, char **argv)
     }
     if (debugLevel == 0 && daemonMode)
 	BecomeDaemon ();
+    if (debugLevel == 0)
+	InitErrorLog ();
     if (debugLevel >= 10)
 	nofork_session = 1;
     /* SUPPRESS 560 */
@@ -157,8 +159,6 @@ main (int argc, char **argv)
 		 pidFile, oldpid);
 	exit (1);
     }
-    if (debugLevel == 0)
-	InitErrorLog ();
 
     if (nofork_session == 0) {
 	/* Clean up any old Authorization files */
-- 
1.5.6.3

